dashboard/app/models/unit.rb

```diff
@@ -1614,10 +1625,13 @@ def summarize_for_rollup(user = nil)
   end
 
   def summarize_for_unit_edit
+    co = course_version&.course_offering
     include_lessons = false
     summary = summarize(include_lessons)
     summary[:lesson_groups] = lesson_groups.map(&:summarize_for_unit_edit)
-    summary[:courseOfferingEditPath] = edit_course_offering_path(course_version.course_offering.key) if course_version
+    summary[:courseOfferingEditPath] = edit_course_offering_path(co&.key) if course_version
+    # Course offering device compatibilites are required to publish student courses that are assignable.
+    summary[:missingRequiredDeviceCompatibilities] = co&.assignable? && co&.get_participant_audience == 'student' && missing_device_compatibility?
```

Comment:
why do we need to check `assignable?` here?

Reply:
This is to match which courses are shown on the Curriculum Catalog page (minus the condition of being published). The ones shown on the catalog page are student assignable published courses.